crates/vim/src/normal/scroll.rs

```diff
@@ -91,12 +91,13 @@ fn scroll_editor(
         s.move_with(|map, selection| {
             let mut head = selection.head();
             let top = top_anchor.to_display_point(map);
+            let starting_column = head.column();
```

Comment:
I'm storing the starting column due to (I think) a bug in `map.clip_point`, which will move the cursor to the right when passed a display point that is past the last line, even if you provide `Bias::Left`.

There's almost surely a better way to fix this root issue, but using the starting_column after `clip_point` ensures that we attempt to go return to the column we started on, which more closely matches neovim behavior. Neovim's behavior is actually more complex than that, but I think that would be a can of worms for this PR.

This fix isn't necessary for the issue I originally opened. However, when I was writing the test for the original issue, I was running into a difference in behavior between the column that the cursor ended up on between neovim and zed. I can revert this change, but I would likely need to add a hack into the new test to re-adjust the column.